src/rvsdg/egglog_optimizer/extraction_rules.rs

```diff
@@ -0,0 +1,204 @@
+use super::BRIL_OPS;
+
+/// These extraction rules are not (yet) used to perform actual extraction.
```

Comment:
I think we need more text here. 

In particular, we do the optimization _during_ extraction, we don't have subset of rules that simply look at the extractions output.